articles/mobile-automation-testing-with-appium/index.md

```diff
@@ -0,0 +1,99 @@
+
+### How Appium works?
+Appium works using the three main components of its architecture (client, server, and end device). When this tool is installed on your computer, it sets up an HTTP server that generates a REST API.
+
+This server collects command requests from the Appium client in JSON format. JSON Wire Protocol plays the role of communicating these requests. These command requests are then executed on the end devices (either Android, Windows, or iOS).
+
+The Appium server uses a test automation framework to execute requests on the user interface of end devices. For example, it uses *XCUI test* to execute commands on iOS. It uses *UI Automator* to execute requests on Android platforms (for newer API versions). After the requests have been executed, the Appium server provides HTTP responses to the client.
+
+Appium works differently on iOS and Android. Let's look at how it works on these two platforms.
+
+#### How Appium works on Android?
+![Appium on Android](/engineering-education/articles/mobile-automation-testing-with-appium/appium-on-android.png)
+[Image Source: Edureka](https://www.edureka.co/blog/content/ver.1556540029/uploads/2019/05/Appium-on-Android-Appium-Architecture-edureka-4.png)
+
+The following is a brief sequence of how Appium works on the Android platform:
+1.The Appium client connects with the HTTP server and uses the JSON Wire Protocol for communication.
+2.The Appium server receives requests from the client component.
+3.The server connects with the mobile automation framework for Android (UI Automator or Selendroid).
+4.Android devices consist of **bootstrap.jar** that plays the role of a [TCP server](https://en.wikipedia.org/wiki/Transmission_Control_Protocol). The **bootstrap.jar** receives the test commands from the Appium server.  
+5.**Bootstrap.jar** uses the UI Automator or Selendroid to execute the action on the Android device.
+6.The test results are then sent to the server, which eventually sends HTTP responses to the Appium client.
+
+#### How Appium works on iOS?
+![Appium on iOS](/engineering-education/articles/mobile-automation-testing-with-appium/appium-on-ios.png)
+[Image Source: Edureka](https://www.edureka.co/blog/content/ver.1556540029/uploads/2019/05/Appium-on-iOS-Appium-Architecture-edureka-4.png)
+
+The following steps show how Appium works on the iOS platform:
+1.The Appium client connects with the HTTP server and uses the JSON Wire Protocol for communication.
+2.The Appium server receives requests from the client component.
+3.The server connects with the mobile automation framework for iOS devices (XCUI test).
+4.iOS devices consist of **bootstrap.js** that plays the role of a TCP server. **Bootstrap.js** receives the test commands from the Appium server.  
+5.**Bootstrap.js** uses the XCUI test to execute the action on the iOS device.
+6.The test results are then sent to the server, which eventually sends HTTP responses to the Appium client.
+

```

Comment:
Also, the working of Appium in Android/iOS could be more elaborative. The whole topic on the working of Appium is not very beginner friendly. If this article is not intended for beginners, specifying the reader level of understanding, at the beginning of the article would be important.